features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/interactiontonextview/InteractionToNextViewMetricResolver.kt

```diff
@@ -0,0 +1,116 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.metric.interactiontonextview
+
+import androidx.annotation.VisibleForTesting
+import com.datadog.android.api.InternalLogger
+
+internal class InteractionToNextViewMetricResolver(
+    private val internalLogger: InternalLogger,
+    private val ingestionValidator: InteractionIngestionValidator = ActionTypeInteractionValidator(),
+    private val lastInteractionIdentifier: LastInteractionIdentifier = TimeBasedInteractionIdentifier()
+) {
+
+    private val lastInteractions = LinkedHashMap<String, InternalInteractionContext>()
+    private val lastViewCreatedTimestamps = LinkedHashMap<String, Long>()
+
+    fun onViewCreated(viewId: String, timestamp: Long) {
+        lastViewCreatedTimestamps[viewId] = timestamp
+        purgeOldEntries()
+    }
+
+    fun onActionSent(context: InternalInteractionContext) {
+        if (ingestionValidator.validate(context)) {
+            lastInteractions[context.viewId] = context
+        }
+        purgeOldEntries()
+    }
+
+    @Suppress("ReturnCount")
+    fun resolveMetric(viewId: String): Long? {
+        purgeOldEntries()
+        val currentViewCreatedTimestamp = lastViewCreatedTimestamps[viewId]
+        if (currentViewCreatedTimestamp == null) {
+            internalLogger.log(
+                InternalLogger.Level.WARN,
+                InternalLogger.Target.MAINTAINER,
+                { "[ViewNetworkSettledMetric] The view was not yet created for this viewId:$viewId" }
+            )
+            return null
+        }
+        val lastPrevViewInteraction = resolveLastInteraction(viewId, currentViewCreatedTimestamp)
+        if (lastPrevViewInteraction != null) {
+            val difference = currentViewCreatedTimestamp - lastPrevViewInteraction.eventCreatedAtNanos
+            if (difference > 0) {
+                return difference
+            } else {
+                internalLogger.log(
+                    InternalLogger.Level.WARN,
+                    InternalLogger.Target.MAINTAINER,
+                    {
+                        "[ViewNetworkSettledMetric] The difference between the last interaction " +
+                            "and the current view is negative for viewId:$viewId"
+                    }
+                )
+                return null
+            }
+        }
+        internalLogger.log(
+            InternalLogger.Level.WARN,
+            InternalLogger.Target.MAINTAINER,
+            { "[ViewNetworkSettledMetric] No previous interaction found for this viewId:$viewId" }
+        )
```

Comment:
question/ Won't we always end up here for the initial view (`ApplicationLaunch`)? If so, this log won't be actionable.

Reply:
totally true....I guess we can condition this log by checking if there's any interaction yet added in the map, if not this is the applaunch. Good point. Actually added a test about this but never thought to condition the log.